test/models/csync_record_test.rb

```diff
@@ -0,0 +1,289 @@
+require 'test_helper'
+
+class CsyncRecordTest < ActiveSupport::TestCase
+  setup do
+    @domain = domains(:shop)
+
+    @csync_record = CsyncRecord.new
+    @csync_record.cdnskey = '257 3 13 mdsswUyr3DPW132mOi8V9xESWE8jTo0dxCjjnopKl+GqJxpVXckHAeF+KkxLbxILfDLUT0rAK9iUzy1L53eKGQ=='
+    @csync_record.action = 'initialized'
+    @csync_record.domain = domains(:shop)
+    @csync_record.last_scan = Time.now
+  end
+
+  def test_domain_must_be_present
+    @csync_record.domain = nil
+    assert_not @csync_record.valid?
+    @csync_record.domain = domains(:shop)
```

Comment:
Is this necessary? Shouldn't it be already set by setup section?

Reply:
Good catch. Changing order of these checks is a good idea.